app/src/main/java/com/gruppo4/ringUp/activities/InstructionsActivity.java

```diff
@@ -0,0 +1,82 @@
+package com.gruppo4.ringUp.activities;
+
+import android.content.Context;
+import android.content.Intent;
+import android.os.Bundle;
+import android.view.View;
+import android.widget.Button;
+import android.widget.Toast;
+
+import androidx.appcompat.app.AppCompatActivity;
+
+import com.gruppo4.ringUp.R;
+import com.gruppo4.ringUp.dialog.SetPasswordListener;
+import com.gruppo4.ringUp.structure.PasswordManager;
+import com.gruppo4.ringUp.dialog.PasswordDialog;
+
+import static com.gruppo4.ringUp.activities.MainActivity.DIALOG_TAG;
+
+/**
+ * The following activity aims to inform the user about what the application is doing and to set the password of his device, necessary for using the app.
+ *
+ * @author Alberto Ursino
+ * @version 1.0
+ * @since 07/01/2020
+ */
+public class InstructionsActivity extends AppCompatActivity implements SetPasswordListener {
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_instructions);
+
+        Button setPassButton = findViewById(R.id.set_pass_button);
+        setPassButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                openPassDialog();
+            }
+        });
+    }
+
+    /**
+     * Method used to, in chronological order:
+     * 1) Saves the personal device password in memory through the {@link PasswordManager} class
+     * 2) Starts the {@link MainActivity}
+     * 3) Closes the {@link InstructionsActivity}, it is no longer necessary
```

Comment:
Well, I think it's better to don't specify WHERE the password is saved because I don't know for sure.
What do you think, in line 44, of write "`Save the personal device password through the {@link PasswordManager#setPassword(String, Context)}`" class?

Reply:
You're calling `PasswordManager.setPassword()`, which calls `PreferencesManager.setString()`, which calls `Editor.putString()`, which writes the password to secondary storage. Your edit is fine, except I would change the last word, "class", with "method".

Reply:
Mmm ok if you say that `Editor.putString()` store datas to the _secondary storage_ I'll trust you.